src/app/reporting/Engine.cpp

```diff
@@ -364,18 +401,24 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
 
     SuccessOrExit(err = reportDataWriter.ReserveBuffer(kReservedSizeForMoreChunksFlag + kReservedSizeForEndOfReportMessage));
 
-    err = BuildSingleReportDataAttributeReportIBs(reportDataBuilder, apReadHandler, &hasMoreChunks);
+    err = BuildSingleReportDataAttributeReportIBs(reportDataBuilder, apReadHandler, &hasMoreChunksForAttributes,
+                                                  &hasEncodedAttributes);
     SuccessOrExit(err);
 
-    err = BuildSingleReportDataEventReports(reportDataBuilder, apReadHandler, &hasMoreChunks);
+    err = BuildSingleReportDataEventReports(reportDataBuilder, apReadHandler, &hasMoreChunksForEvents, &hasEncodedEvents);
     SuccessOrExit(err);
 
-    SuccessOrExit(reportDataBuilder.GetError());
+    if (!hasEncodedAttributes && !hasEncodedEvents && (hasMoreChunksForAttributes || hasMoreChunksForEvents))
```

Comment:
I wonder whether we should set
```
    bool hasMoreChunks = hasMoreChunksForAttributes || hasMoreChunksForEvents;
```
instead of copy/pasting this expression all down this function.